src/material-experimental/mdc-slider/slider-adapter.ts

```diff
@@ -8,131 +8,147 @@
 
 import {SpecificEventListener, EventType} from '@material/base';
 import {MDCSliderAdapter, Thumb, TickMark} from '@material/slider';
+import {MatSlider} from './slider';
 
 export class SliderAdapter implements MDCSliderAdapter {
-  hasClass(className: string): boolean {
-    throw Error('Method not implemented.');
+  constructor(private readonly _delegate: MatSlider) {}
+  hasClass = (className: string): boolean => {
+    return this._delegate._hostElement.classList.contains(className);
   }
-  addClass(className: string): void {
-    throw Error('Method not implemented.');
+  addClass = (className: string): void => {
```

Comment:
Why were all of these changed to arrow functions?

Reply:
I'm honestly not sure. For some reason, unless I do this, these functions do not continue to exist when the slider adapter is used

Reply:
You should look to see if the MDC implementation is clobbering your methods, and then doing this clobbers them back. If so, we should fix MDC. Doing this defeats the point of making a class because you end up with one function instance per class instance instead of a single function instance shared across all class instances via the prototype.